nemoguardrails/library/autoalign/actions.py

```diff
@@ -0,0 +1,291 @@
+            headers=headers,
+            json=request_body,
+        ) as response:
+            if response.status != 200:
+                raise ValueError(
+                    f"AutoAlign call failed with status code {response.status}.\n"
+                    f"Details: {await response.text()}"
+                )
+            async for line in response.content:
+                line_text = line.strip()
+                if len(line_text) > 0:
+                    resp = json.loads(line_text)
+                    guardrails_configured.append(resp)
+            processed_response = process_autoalign_output(
+                guardrails_configured, show_toxic_phrases
+            )
+    return processed_response
+
+
+async def autoalign_factcheck_infer(
+    request_url: str,
+    text: str,
+    documents: List[str],
+):
+    """Checks the facts for the text using the given documents and provides a fact-checking score"""
+    api_key = os.environ.get("AUTOALIGN_API_KEY")
+    if api_key is None:
+        raise ValueError("AUTOALIGN_API_KEY environment variable not set.")
+    headers = {"x-api-key": api_key}
+    request_body = {
+        "prompt": text,
+        "documents": documents,
+    }
+    async with aiohttp.ClientSession() as session:
+        async with session.post(
+            url=request_url,
+            headers=headers,
+            json=request_body,
+        ) as response:
+            if response.status != 200:
+                raise ValueError(
+                    f"AutoAlign call failed with status code {response.status}.\n"
+                    f"Details: {await response.text()}"
+                )
+            async for line in response.content:
+                resp = json.loads(line)
+                if resp["task"] == "factcheck":
+                    return float(resp["response"][17:])
+    return 1.0
+
+
+@action(name="autoalign_input_api")
+async def autoalign_input_api(
+    llm_task_manager: LLMTaskManager,
+    context: Optional[dict] = None,
+    show_autoalign_message: bool = True,
+    show_toxic_phrases: bool = False,
+):
+    """Calls AutoAlign API for the user message and guardrail configuration provided"""
+    user_message = context.get("user_message")
+    autoalign_config = llm_task_manager.config.rails.config.autoalign
+    autoalign_api_url = autoalign_config.parameters.get("endpoint")
+    if not autoalign_api_url:
+        raise ValueError("Provide the autoalign endpoint in the config")
+    task_config = getattr(autoalign_config.input, "guardrails_config")
+    if not task_config:
+        raise ValueError("Provide the guardrails and their configuration")
+    text = user_message
+
+    autoalign_response = await autoalign_infer(
+        autoalign_api_url, text, task_config, show_toxic_phrases
+    )
+    if autoalign_response["guardrails_triggered"] and show_autoalign_message:
+        print(
+            f"AutoAlign on Input: {autoalign_response['combined_response']}",
+        )
```

Comment:
Use `log.info` rather than `print`.

Reply:
We have used log.warning instead of log.info for user visibility ( 2b446091693e44cb629d0cd79223a9519df1cdd1 )